### PR TITLE
[4.x]  Simplifying queuing internal middleware

### PR DESCRIPTION
**Pull request by adriansuter**
This PR addresses #2762.

Not sure, if the test cases test everything that needs to be tested.

### CODE REVIEW

**Comment by l0gicgate** [comment]
Looks good to me @adriansuter, great work! As for the tests, the middleware test suite was already covering most of this anyway but it's good you went above and beyond with it! We just need to update the docs to reflect those changes. I'm going to merge.

**Review by l0gicgate — APPROVED** [review]

File: Slim/App.php [hunk 1]
```diff
@@ -101,6 +103,46 @@ public function addMiddleware(MiddlewareInterface $middleware): self
...
+     * @param bool $displayErrorDetails
+     * @param bool $logErrors
+     * @param bool $logErrorDetails
+     *
+     * @return ErrorMiddleware
+     */
+    public function addErrorMiddleware(
+        bool $displayErrorDetails,
+        bool $logErrors,
+        bool $logErrorDetails
```
Comment (shadowhand) [inline Slim/App.php:133]: In the interest of simplification, should these default to `true`?

Comment (l0gicgate) [reply Slim/App.php:133]: I think we need to keep the signature the same as the middleware implementation. Wasn't there someone claiming that true defaults were the root of all evil here not long ago? @akrabat 😂

Comment (adriansuter) [reply Slim/App.php:133]: Also the defaults would depend on the environment. In development a user might want to have all set to `true`, but in production one should not display error details (security issue). So I think, we should keep it as is. So the user has to think about these flags.

Comment (akrabat) [reply Slim/App.php:133]: Defaults are difficult  :)

Comment (akrabat) [reply Slim/App.php:133]: Do we actually need `logErrorDetails`? i.e. what's the use-case for logging errors, but not the details?

Comment (adriansuter) [reply Slim/App.php:133]: @akrabat Probably disk memory limitation concerns?

**Pull request merged**